servers/zts/src/main/java/com/yahoo/athenz/zts/cert/impl/DynamoDBClientFetcher.java

```diff
@@ -0,0 +1,172 @@
+        return AmazonDynamoDBClientBuilder.standard()
+                .withCredentials(credentialsProvider)
+                .withRegion(dynamoDBClientSettings.getRegion())
+                .build();
+    }
+
+    private AWSCredentialsProvider getCredentials(DynamoDBClientSettings dynamoDBClientSettings,
+                                                  ZTSClientNotificationSender ztsClientNotificationSender) {
+        try {
+            KeyRefresher keyRefresher = Utils.generateKeyRefresher(
+                    dynamoDBClientSettings.getTrustStore(),
+                    dynamoDBClientSettings.getTrustStorePassword(),
+                    dynamoDBClientSettings.getCertPath(),
+                    dynamoDBClientSettings.getKeyPath());
+            keyRefresher.startup();
+
+            SSLContext sslContext = Utils.buildSSLContext(keyRefresher.getKeyManagerProxy(),
+                    keyRefresher.getTrustManagerProxy());
+
+            String externalId = System.getProperty(ZTS_PROP_DYNAMODB_EXTERNAL_ID, null);
+            String minExpiryTimeStr = System.getProperty(ZTS_PROP_DYNAMODB_MIN_EXPIRY_TIME, "");
+            String maxExpiryTimeStr = System.getProperty(ZTS_PROP_DYNAMODB_MAX_EXPIRY_TIME, "");
+            Integer minExpiryTime = minExpiryTimeStr.isEmpty() ? null : Integer.parseInt(minExpiryTimeStr);
+            Integer maxExpiryTime = maxExpiryTimeStr.isEmpty() ? null : Integer.parseInt(maxExpiryTimeStr);
+
+            return new AWSCredentialsProviderImpl(
+                    dynamoDBClientSettings.getZtsURL(),
+                    sslContext,
+                    dynamoDBClientSettings.getDomainName(),
+                    dynamoDBClientSettings.getRoleName(),
+                    externalId,
+                    minExpiryTime,
+                    maxExpiryTime,
+                    ztsClientNotificationSender);
+
+        } catch (Exception ex) {
+            LOGGER.error("Failed to get AWS Temporary credentials: {}", ex.getMessage());
+        }
+        return null;
+    }
+
+    public static class DynamoDBClientSettings {
+        private String certPath;
+        private String domainName;
+        private String roleName;
+        private String trustStore;
+        private String trustStorePassword;
+        private String ztsURL;
+        private String region;
+        private String keyPath;
+
+        public DynamoDBClientSettings() {
+            keyPath = System.getProperty(ZTS_PROP_DYNAMODB_KEY_PATH, "");
+            certPath = System.getProperty(ZTS_PROP_DYNAMODB_CERT_PATH, "");
+            domainName = System.getProperty(ZTS_PROP_DYNAMODB_DOMAIN, "");
+            roleName = System.getProperty(ZTS_PROP_DYNAMODB_ROLE, "");
+            trustStore = System.getProperty(ZTS_PROP_DYNAMODB_TRUSTSTORE, "");
+            region = System.getProperty(ZTS_PROP_DYNAMODB_REGION, "");
+            trustStorePassword = System.getProperty(ZTS_PROP_DYNAMODB_TRUSTSTORE_PASSWORD, "");
```

Comment:
we should not read the password directly from the property even if it's only for the truststore. The factory is given a PrivateKeyStore object that should be used to fetch any passwords. I think we should have some examples in the ZTSUtils class

Reply:
I did the same as when reading the password using the keyStore in JDBCSSHRecordStoreFactory.
However, I wasn't able to test it yet as I'm not sure how to actually place a value in the keyStore before retrieving it. I'm going over the CKMS guide to learn about it.